src/uipath/_services/attachments_service.py

```diff
@@ -0,0 +1,780 @@
+                    "PUT", upload_uri, headers=headers, files={"file": file}
+                )
+            else:
+                request("PUT", upload_uri, headers=headers, files={"file": file})
+                
+        return attachment_key
+
+    @traced(
+        name="attachments_upload_from_memory",
+        run_type="uipath",
+        input_processor=_upload_attachment_input_processor,
+    )
+    def upload_from_memory(
+        self,
+        *,
+        name: str,
+        content: Union[str, bytes],
+        folder_key: Optional[str] = None,
+        folder_path: Optional[str] = None,
+    ) -> uuid.UUID:
+        """Upload content from memory as an attachment.
+
+        This method uploads content directly from memory to UiPath as an attachment.
+
+        Args:
+            name (str): The name of the attachment file.
+            content (Union[str, bytes]): The content to upload (string or bytes).
+            folder_key (Optional[str]): The key of the folder. Override the default one set in the SDK config.
+            folder_path (Optional[str]): The path of the folder. Override the default one set in the SDK config.
+
+        Returns:
+            uuid.UUID: The UUID of the created attachment.
+
+        Raises:
+            Exception: If the upload fails.
+
+        Examples:
+            ```python
+            from uipath import UiPath
+
+            client = UiPath()
+
+            # Upload content from memory
+            attachment_key = client.attachments.upload_from_memory(
+                name="notes.txt",
+                content="This is a text file content",
+            )
+            print(f"Uploaded attachment with key: {attachment_key}")
+            ```
+        """
+        spec = self._create_attachment_and_retrieve_upload_uri_spec(
+            name=name,
+            folder_key=folder_key,
+            folder_path=folder_path,
+        )
+
+        result = self.request(
+            spec.method,
+            url=spec.endpoint,
+            params=spec.params,
+            headers=spec.headers,
+            json=spec.json,
+        ).json()
+
+        # Get the ID from the response and convert to UUID
+        attachment_key = uuid.UUID(result["Id"])
+
+        upload_uri = result["BlobFileAccess"]["Uri"]
+        headers = {
+            key: value
+            for key, value in zip(
+                result["BlobFileAccess"]["Headers"]["Keys"], result["BlobFileAccess"]["Headers"]["Values"], strict=False
+            )
+        }
+
+        # Convert string to bytes if needed
+        if isinstance(content, str):
+            content = content.encode("utf-8")
+
+        if result["BlobFileAccess"]["RequiresAuth"]:
+            self.request("PUT", upload_uri, headers=headers, content=content)
+        else:
+            request("PUT", upload_uri, headers=headers, content=content)
+            
+        return attachment_key
+
+    @traced(
+        name="attachments_upload_from_memory",
```

Comment:
this name will appear in the traces (user friendly-name), it doesn't need to be the exact name of the method, eg in this case it should just be "attachments_upload"  (similar to the other services)